### PR TITLE
✨ Add tests for graphQL costs

### LINKED ISSUE #1623: BUG: Fixer high GitHub API usage
#1566 increases the cost/complexity of the GraphQL queries. The query cost goes up ~20x due to that PR. We should fix this before the next release. 2-part solution here:

1. Decrease the number of `associatedPullRequests` to 1, since for every commit we are only interested in a single PR (the one used to submit the commit). Doing this brings back the cost to original, so ~20x savings.
2. Add some tests to verify the cost of our GraphQL queries. Any future issues can be caught using these tests. 

I would say, we block the next release on this issue since IMO it's important to fix this.

### PR DESCRIPTION
**Pull request by azeemshaikh38**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Closes #1623 

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

### CODE REVIEW

**Review by naveensrinivasan — APPROVED** [review]
Awesome! Thanks

**Review by laurentsimon — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -261,7 +261,7 @@ test: $(test-targets)
 unit-test: ## Runs unit test without e2e
 	# Run unit tests, ignoring e2e tests
 	# run the go tests and gen the file coverage-all used to do the integration with codecov
-	go test -race -covermode=atomic  -coverprofile=unit-coverage.out `go list ./... | grep -v e2e`
+	SKIP_GINKGO=1 go test -race -covermode=atomic  -coverprofile=unit-coverage.out `go list ./...`
```
Comment (06kellyjac) [inline Makefile:264]: Is there a reason these e2e tests couldn't sit in the e2e dir?

Comment (naveensrinivasan) [reply Makefile:264]: @06kellyjac  Nice suggestion, Thanks. Do you want to do a PR for the change?

Comment (azeemshaikh38) [reply Makefile:264]: The tests added in this PR rely on some internal (non-exposed) fields. So, had to create the tests in the same package as it was testing.

Comment (06kellyjac) [reply Makefile:264]: fair enough, thanks for the details

**Pull request merged**